lessonPlans/views.py

```diff
@@ -32,6 +32,14 @@ def topic_detail(request, topic_id):
     try:
         topic = Topic.objects.get(pk = topic_id)
         description = topic.description
+        current_user = request.user
+        plans = []
+
+        for plan in LessonPlan.objects.all():
+            if current_user in plan.assigned_users.all():
+                for plan_topic in plan.topics.filter(name = topic.name):
```

Comment:
Do we need to use `filter` here? Can we just check if the required topic belongs to the lesson plan?

Reply:
I suppose this is a more efficient way of doing it though I'm not 100% sure, topics is an array and we need to check if the particular topic is inside that array so we'll have to add an additional if, the filter does that for us but if you prefer it that way I can change.

Reply:
I think `plan.topics.filter(name = topic.name)` would either return an empty array or an array with one element. So, the `filter` and `for` in this case seem trivial to me.

Reply:
Oh my bad 😅, thanks for pointing that out